src/PrimaryGeneratorActionMessenger.cpp

```diff
@@ -149,7 +182,32 @@ void PrimaryGeneratorActionMessenger::CheckIfIsotope()
 
 }
 
-void PrimaryGeneratorActionMessenger::CheckIfBeam()
+bool PrimaryGeneratorActionMessenger::CheckIfRun()
+{
+  G4int nRun = DetectorConstruction::GetInstance()->GetRunNumber();
+   if(  nRun != 0){ 
+     fPrimGen->SetSourceTypeInfo("run");
+     return true;
+   } else {
+     return false;
+   }
+
+}
```

Comment:
You don't have to use `else` here:
bool PrimaryGeneratorActionMessenger::CheckIfRun()
{
  G4int nRun = DetectorConstruction::GetInstance()->GetRunNumber();
   if(  nRun != 0)
   { 
     fPrimGen->SetSourceTypeInfo("run");
     return true;
   }
     return false;
}